### PR TITLE
Fixes #32070 - Updating welcome pages to use react components

### PR DESCRIPTION
**Pull request by adiabramovitch**

### CODE REVIEW

File: app/views/common_parameters/welcome.html.erb [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 <% content_for(:title, _("Global Parameters")) %>
-<div class="blank-slate-pf">
-  <div class="blank-slate-pf-icon">
-    <%= icon_text("list-ul", "", :kind => "fa") %>
-  </div>
-  <h1><%= _('Global Parameters') %></h1>
-  <p><%= _("Foreman allows to define a hierarchy of parameter inheritance, where global parameters accessible from any manifest.") %></p>
-  <p><%= link_to _('Learn more about this in the documentation.'), documentation_url("global-parameters")%></p>
-  <div class="blank-slate-pf-main-action">
-    <%= new_link(_("Create Global Parameter"), :class => 'btn-lg') %>
```
Comment (tbrisker) [inline app/views/common_parameters/welcome.html.erb:10]: Note that `new_link` checks if the user is permitted to create the resource before showing the link - https://github.com/theforeman/foreman/blob/develop/app/helpers/application_helper.rb#L145
So when switching this to use react, we should only pass the `create` action if the user has the correct permission (or they will get a link that they cannot use). 
This is true for many of the other pages as well. For the custom components, you can add a prop that defines if the user can create and decide according to that if the link should be displayed.

File: app/views/auth_sources/_welcome.html.erb [hunk 2]
```diff
@@ -1,19 +1,6 @@
...
-    <%= new_link(_("Create LDAP Authentication Source"), :class => 'btn-lg',  :controller => :auth_source_ldaps, :action => :new) %>
-  </div>
-</div>
\ No newline at end of file
+
+<%  content_for(:title, _("Authentication Sources")) %> 
+ 
+<%= react_component('WelcomeAuthSource') %> 
+
+
```
Comment (tbrisker) [inline app/views/auth_sources/_welcome.html.erb]: Looks like an extra blank line here, and there are several other cases where you added or removed blank lines. 
While not crucial, it's good to avoid too many whitespace changes and try to keep the spacing consistent. Some IDEs even have options to ensure that all extra whitespace is cleaned up before saving automatically.
In ruby files the ruby linter (rubocop) will yell at you in such cases, but we don't lint `.erb` files.

File: app/views/media/welcome.html.erb [hunk 3]
```diff
@@ -16,4 +16,4 @@
   <p>
   <%= (_("You may also associate one or more operating systems with this medium or alternatively set this up later on the %s page.") % link_to(_("Operating systems"), operatingsystems_path)).html_safe %>
   </p>
-</div>
+</div>
```
Comment (tbrisker) [inline app/views/media/welcome.html.erb]: looks like all that changed here was removing the newline at the end of the file?

File: app/views/ptables/welcome.html.erb [hunk 4]
```diff
@@ -1,6 +1,5 @@
-<% content_for(:title, _("Partition Tables")) %>
+<% content_for(:title, _("Partition")) %>
```
Comment (tbrisker) [inline app/views/ptables/welcome.html.erb]: Why was this changed?

**Review by tbrisker — CHANGES_REQUESTED** [review]
I haven't gone over the JS code yet, since it seems there are many linting changes needed to it, but see inline comments regarding the ruby code. 
You can run eslint locally with `npm run lint` to see all of the linting issues.

File: app/views/architectures/welcome.html.erb [hunk 5]
```diff
@@ -1,16 +1,4 @@
...
-    <%= _("Each entries represent a particular hardware architecture, most commonly <b>x86_64</b> or <b>i386</b>. Foreman also supports the Solaris operating system family, which includes <b>sparc</b> based systems.").html_safe %></br>
-    <%= _("Each architecture can also be associated with more than one operating system and a selector block is provided to allow you to select valid combinations.") %>
-  </p>
-  <p><%= link_to _('Learn more about this in the documentation.'), documentation_url("4.4.5Architectures")%></p>
-  <div class="blank-slate-pf-main-action">
-      <%= new_link(_("Create Architecture"), {}, { :class => "btn-lg" }) %>
-  </div>
-</div>
+
+<%= react_component('WelcomeArchitecture') %>
```
Comment (ekohl) [inline app/views/architectures/welcome.html.erb]: Small nit: we generally use line endings at the end of files. This has the benefit that if another line is added, it doesn't show a diff for this line. Most editors can be configured to ensure it's present. You can see it here as the circle with dash icon, but it's also an issue in other files.

File: webpack/assets/javascripts/react_app/components/ConfigReports/Welcome.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import EmptyState from '../common/EmptyState';
+import {foremanUrl, getManualURL} from '../../common/helpers'
+
+export const WelcomeConfigReports = () => {
+const description  = <> 
+{__("You don't seem to have any reports.") } <br/> 
+{__("If you wish to configure Puppet to forward it reports to Foreman, please follow")}
+<a href = {getManualURL("3.5.4PuppetReports")}> {__("setting up reporting")} </a>
+{__(" and ")}
+<a href = {"http://theforeman.org/projects/foreman/wiki/Mail_Notifications"}> {__("e-mail reporting")} </a>
```
Comment (ekohl) [inline webpack/assets/javascripts/react_app/components/ConfigReports/Welcome.js]: I think we have a link helper to link to the wiki. In https://github.com/theforeman/foreman/pull/8184 I attempted to use it. Please consider doing to right from the start here.

If not, at least use HTTPS.
```suggestion
<a href = {"https://theforeman.org/projects/foreman/wiki/Mail_Notifications"}> {__("e-mail reporting")} </a> 
```

File: app/views/common_parameters/welcome.html.erb [hunk 7]
```diff
@@ -1,12 +1,13 @@
...
-</div>
+
+<% description = _("Foreman allows to define a hierarchy of parameter inheritance, where global parameters accessible from any manifest.")%>
+<%= react_component('EmptyState', {
+                                    :icon => 'list-ul',
+                                    :iconType => 'fa',
+                                    :header => _('Global Parameters'),
+                                    :description => description,
+                                    :documentation => { :url => documentation_url("global-parameters") },
+                                    :action => { :title => _("Create Global Parameter"), :url => new_common_parameter_path }
```
Comment (tbrisker) [inline app/views/common_parameters/welcome.html.erb]:
```suggestion
                                    (:action => { :title => _("Create Global Parameter"), :url => new_common_parameter_path } if authorized_for(action: :new))
```

File: app/views/architectures/welcome.html.erb [hunk 8]
```diff
@@ -1,16 +1,3 @@
...
-    <%= _("Before you proceed to using Foreman you should provide information about one or more architectures.") %></br>
-    <%= _("Each entries represent a particular hardware architecture, most commonly <b>x86_64</b> or <b>i386</b>. Foreman also supports the Solaris operating system family, which includes <b>sparc</b> based systems.").html_safe %></br>
-    <%= _("Each architecture can also be associated with more than one operating system and a selector block is provided to allow you to select valid combinations.") %>
-  </p>
-  <p><%= link_to _('Learn more about this in the documentation.'), documentation_url("4.4.5Architectures")%></p>
-  <div class="blank-slate-pf-main-action">
-      <%= new_link(_("Create Architecture"), {}, { :class => "btn-lg" }) %>
-  </div>
-</div>
+<%= react_component('WelcomeArchitecture', {can_create: authorized_for(cation: :new)}) %>
```
Comment (tbrisker) [inline app/views/architectures/welcome.html.erb]: typo:
```suggestion
<%= react_component('WelcomeArchitecture', {can_create: authorized_for(action: :new)}) %>
```

File: app/views/fact_values/welcome.html.erb [hunk 9]
```diff
@@ -1,12 +1,13 @@
...
+<%= react_component('EmptyState', {
+                                    :icon => 'list-alt',
+                                    :iconType => 'fa',
+                                    :header => _('Fact Values'),
+                                    :description => description,
+                                    :documentation => { :url => documentation_url("3.5.5FactsandtheENC") },
+                                  }) %>
+
+
+
```
Comment (tbrisker) [inline app/views/fact_values/welcome.html.erb]: lots of empty lines here

File: app/views/locations/welcome.html.erb [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<% content_for(:title, _("Locations")) %>
+
+<%= react_component('WelcomeLoc') %>
```
Comment (tbrisker) [inline app/views/locations/welcome.html.erb]: This is no longer needed since you dropped it in your other PR (same for organizations and the associated components)

File: app/views/media/welcome.html.erb (hunk 3 above)
Comment (tbrisker) [reply app/views/media/welcome.html.erb]: still all that changed is the newline here, if you aren't changing this file than don't change the whitespace as it will just confuse the history.

File: webpack/assets/javascripts/react_app/components/Architectures/Welcome.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { translate as __ } from '../../common/I18n';
+import EmptyState from '../common/EmptyState';
+import { foremanUrl } from '../../common/helpers';
+
+export const WelcomeArchitecture = ({can_create}) => {
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/Architectures/Welcome.js]: Looks like this should be:
```suggestion
export const WelcomeArchitecture = ({ canCreate }) => {
```
and you'll also need to change it in the erb and in the other components

**Review by tbrisker — CHANGES_REQUESTED** [review]

**Comment by adiabramovitch** [comment]
[test katello]

File: webpack/assets/javascripts/react_app/components/Architectures/Welcome.js [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    <>
+      <p>
+        {__(
+          'Before you proceed to using Foreman you should provide information about one or more architectures.'
+        )}
+        <br />
+        {__(
+          'Each entries represent a particular hardware architecture, most commonly'
+        )}
+        <b>{__(' x86_64')}</b> {__('or')} <b>{__('i386.')}</b> <br />
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/Architectures/Welcome.js]: This (and other similar changes) could break the translations in different languages where the sentence structure is different - the translators will not see the full sentence and will not be able to reorder it to match the correct sentence structure in their language.
You would probably need to include the tags inside the translated string and use `dangerouslySetInnerHTML` function to get the translated string converted to HTML.
It would be ideal if you could keep as many of the strings as possible identical to what they were previously, so that existing translations can still be used.

File: webpack/assets/javascripts/react_app/components/Architectures/Welcome.js [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+import { foremanUrl } from '../../common/helpers';
+
+export const WelcomeArchitecture = ({ canCreate }) => {
+  const action = canCreate && {
+    title: __('Create Architecture'),
+    url: foremanUrl('/architectures/new'),
+  };
+  const description = (
+    <>
+      <p>
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/Architectures/Welcome.js]: There is no need for both `<>` and `<p>` - `<>` is needed when you want to group several elements together without wrapping them in another element.

File: webpack/assets/javascripts/react_app/components/Architectures/Welcome.js [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+        {__(
+          'Before you proceed to using Foreman you should provide information about one or more architectures.'
+        )}
+        <br />
+        {__(
+          'Each entries represent a particular hardware architecture, most commonly'
+        )}
+        <b>{__(' x86_64')}</b> {__('or')} <b>{__('i386.')}</b> <br />
+        Foreman also supports the Solaris operating system family, which
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/Architectures/Welcome.js]: You will need to add a couple of strings to https://github.com/theforeman/foreman/blob/develop/.eslintrc#L13 so the spell checker doesn't complain.
Also, this string is not marked for translation.

**Review by tbrisker — CHANGES_REQUESTED** [review]
We're getting there ;) a few more comments inline.

**Comment by adiabramovitch** [comment]
[test foreman]

**Comment by Ron-Lavi** [comment]
@rabajaj0509 can you take a look? thanks

File: app/views/report_templates/welcome.html.erb [hunk 15]
```diff
@@ -1,12 +1,13 @@
...
-  <h1><%= _('Report Templates') %></h1>
-  <p><%= _("The Report Templates provides a way to create textual reports out of Foreman data. When the template is rendered, it loads data from database and renders the result. Based on the template, the output format can be csv, json, yaml or any custom textual format") %></p>
-  <p><%= link_to _('Learn more about this in the documentation.'), documentation_url("4.11ReportTemplates")%></p>
-  <div class="blank-slate-pf-main-action">
-    <%= new_link(_("Create Report Template"), :class => 'btn-lg') %>
-  </div>
-</div>
+
+<% description = _("The Report Templates provides a way to create textual reports out of Foreman data. When the template is rendered, it loads data from database and renders the result. Based on the template, the output format can be csv, json, yaml or any custom textual format")%>
+<% action = { :title => _("Create Report Template"), :url => new_report_template_path } if authorized_for(:action => :new)%>
```
Comment (Ron-Lavi) [inline app/views/report_templates/welcome.html.erb:3]: still waiting to see if the integration test passes...
if not, you may try another selector, locally it worked for me:
```ruby
    page.find('a', text: 'Create Report Template').click
    # assert page.has_link?('Create Report Template')

    # click_link 'Create Report Template'
```

File: test/integration/middleware_test.rb [hunk 16]
```diff
@@ -31,7 +31,7 @@ class MiddlewareIntegrationTest < ActionDispatch::IntegrationTest
     test 'it is added Content-Security-Policy on welcome pages' do
       Domain.stubs(:any?).returns(false)
       visit '/domains'
-      assert page.has_content? 'Learn more about this in the documentation.'
+      # assert page.has_content? 'Learn more about this in the documentation.'
```
Comment (Ron-Lavi) [inline test/integration/middleware_test.rb]: oops? if it didn't work try to use another capybara selector :D

File: webpack/assets/javascripts/react_app/common/helpers.js [hunk 17]
```diff
@@ -161,6 +161,7 @@ export const stringIsPositiveNumber = value => {
  * @param {String} section - section id for foreman documetation
  */
 export const getManualURL = section => foremanUrl(`/links/manual/${section}`);
+export const getWikiURL = section => foremanUrl(`/links/wiki/${section}`);
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/common/helpers.js:164]: nice **Adi**tion @adiabramovitch !!

File: webpack/assets/javascripts/react_app/components/Architectures/Welcome.js [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+      {<div dangerouslySetInnerHTML={{ __html: content }} />}
+      {__(
+        'Each architecture can also be associated with more than one operating system and a selector block is provided to allow you to select valid combinations.'
+      )}
+    </>
+  );
+
+  return (
+    <EmptyState
+      icon="building "
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/Architectures/Welcome.js]: nit, extra space?

File: webpack/assets/javascripts/react_app/components/AuthSource/Welcome.js [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+    </>
+  );
+  const action = canCreate && {
+    title: __('Create LDAP Authentication Source'),
+    url: foremanUrl('auth_source_ldaps/new'),
+  };
+
+  return (
+    <EmptyState
+      icon="users "
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/AuthSource/Welcome.js]: extra space? or is it in purpose and I am missing something lol

File: webpack/assets/javascripts/react_app/components/Architectures/Welcome.js [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+  };
+  const content = __(`Each entries represent a particular hardware architecture, most commonly <b>x86_64</b> or <b>i386</b>.
+  Foreman also supports the Solaris operating system family, which includes <b>sparc</b> based systems.`);
+  const description = (
+    <>
+      {__(
+        'Before you proceed to using Foreman you should provide information about one or more architectures.'
+      )}
+      <br />
+      {<div dangerouslySetInnerHTML={{ __html: content }} />}
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/Architectures/Welcome.js]: I don't think we need to insert it dangerously.
can we just wrap that content with <p> or something? or is the `EmptyState` component doing problems?

File: webpack/assets/javascripts/react_app/components/ConfigReports/Welcome.js [hunk 21]
```diff
@@ -0,0 +1,27 @@
...
+export const WelcomeConfigReports = () => {
+  const content = __(`If you wish to configure Puppet to forward it reports to Foreman, 
+  please follow <a href=${getManualURL(
+    '3.5.4PuppetReports'
+  )}> setting up reporting   </a> and
+  <a href=${getWikiURL('Mail_Notifications')}> e-mail reporting </a>`);
+  const description = (
+    <>
+      {__("You don't seem to have any reports.")} <br />
+      <div dangerouslySetInnerHTML={{ __html: content }} />
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/ConfigReports/Welcome.js:15]: here as well

File: webpack/assets/javascripts/react_app/components/Enviroments/Welcome.js [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+    '4.2.2Classes'
+  )}>Puppet classes and environment importer.</a>`);
+  return (
+    <EmptyState
+      icon="th "
+      iconType="fa"
+      header={__('Environments')}
+      description={
+        <>
+          <div dangerouslySetInnerHTML={{ __html: content }} />
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/Enviroments/Welcome.js]: and here :D

File: webpack/assets/javascripts/react_app/components/Architectures/Welcome.js (hunk 20 above)
Comment (tbrisker) [reply webpack/assets/javascripts/react_app/components/Architectures/Welcome.js]: It is needed because the string contains html and we can't break it into partial strings since that will break the translation. The same for the rest of the strings. We came up with this solution since the sentences can't be broken up due to the way translations work.

File: test/integration/middleware_test.rb (hunk 16 above)
Comment (tbrisker) [reply test/integration/middleware_test.rb]: we're trying to figure out the test failures right now, this is still WIP

File: app/views/report_templates/welcome.html.erb (hunk 15 above)
Comment (tbrisker) [reply app/views/report_templates/welcome.html.erb:3]: Ah, that makes sense actually - `.find` (and all other [`Capybara::Node::Finders`](https://www.rubydoc.info/github/jnicklas/capybara/Capybara/Node/Finders)) continuously retries to find the element until `default_max_wait_time` is reached, while [`Capybara::Node::Matchers`](https://www.rubydoc.info/github/jnicklas/capybara/Capybara/Node/Matchers) like `has_link?` only try once and fail if the element isn't there (such as when the check is done before all of the react code finished rendering). The downside of using finders and not matchers is that they can be much slower, especially when they fail - since they will always wait [20 seconds](https://github.com/theforeman/foreman/blob/develop/test/integration_test_helper.rb#L32) before failing.
In the case of the first check on a react page it makes sense to use a finder, though in this case we could use 
```
assert page.find_link('Create Report Template')
```
to maintain the same behaviour (i.e. failing if the link isn't there).
This comment should be have been made on the test though 😉

File: webpack/assets/javascripts/react_app/components/Architectures/Welcome.js (hunk 20 above)
Comment (tbrisker) [reply webpack/assets/javascripts/react_app/components/Architectures/Welcome.js]: @laviro please read the comment regarding this in my previous review - https://github.com/theforeman/foreman/pull/8418#discussion_r609429894

File: app/views/organizations/welcome.html.erb [hunk 23]
```diff
@@ -0,0 +1,3 @@
+<% content_for(:title, _("Organizations")) %>
+
+<%= react_component('WelcomeOrg') %>
```
Comment (tbrisker) [inline app/views/organizations/welcome.html.erb]: I think this is a leftover from the previous version of this commit :)

File: app/views/usergroups/welcome.html.erb [hunk 24]
```diff
@@ -1,16 +1,19 @@
...
+<%= react_component('EmptyState', {
+                                    :icon => 'project',
+                                    :iconType => 'pf',
+                                    :header => _('User Groups'),
+                                    :description => description,
+                                    :action => action
+                                  }) %>
+
+
+
```
Comment (tbrisker) [inline app/views/usergroups/welcome.html.erb]: Lots of extra blank lines here :)

File: test/integration/org_admin_js_test.rb [hunk 25]
```diff
@@ -142,7 +142,7 @@ def test_org_admins_can_work_with_resources_in_their_organization
       visit domains_path
       wait_for_ajax
 
-      click_link 'Create Domain'
+      click_button 'Create Domain'
```
Comment (tbrisker) [inline test/integration/org_admin_js_test.rb]: I think there should be at least one domain in the db so these should stay `click_link`, it's the link on the top of the domain table and not the welcome page? (same for the others below)
Previously only one test was failing, that is the one we need to fix since it is the one that goes to the welcome page (i don't remember which one it was though)

File: webpack/assets/javascripts/react_app/components/Architectures/Welcome.js [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+    url: foremanUrl('/architectures/new'),
+  };
+  const content = __(`Each entries represent a particular hardware architecture, most commonly <b>x86_64</b> or <b>i386</b>.
+  Foreman also supports the Solaris operating system family, which includes <b>sparc</b> based systems.`);
+  const description = (
+    <>
+      {__(
+        'Before you proceed to using Foreman you should provide information about one or more architectures.'
+      )}
+      <br />
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/Architectures/Welcome.js]: With the new `<div>` I think you don't need an extra `<br />` since the div would break the line anyway

File: webpack/assets/javascripts/react_app/components/AuthSource/Welcome.js [hunk 27]
```diff
@@ -0,0 +1,48 @@
...
+
+export const WelcomeAuthSource = ({ canCreate }) => {
+  const content1 = __(`The authentication process currently requires an LDAP provider, such as <em>FreeIPA</em>, <em>OpenLDAP</em> or <em>Microsoft's Active Directory</em>.
+  <a href=${getManualURL(
+    '4.1.1LDAPAuthentication'
+  )}>Learn more about LDAP authentication in the documentation.</a>`);
+  const content2 = __(`Foreman can use External service for user information and authentication. <br/>
+  <a href=${getManualURL(
+    '5.7ExternalAuthentication'
+  )}>Learn more about External authentication in the documentation.</a>`);
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/AuthSource/Welcome.js]: This layout looks a bit odd... I'm not sure if eslint will permit it, but something like this would probably be easier to read:
```suggestion
  const content1 = __(`The authentication process currently requires an LDAP provider,
     such as <em>FreeIPA</em>, <em>OpenLDAP</em> or <em>Microsoft's Active Directory</em>.
    <a href=${getManualURL('4.1.1LDAPAuthentication')}>Learn more about LDAP authentication in the documentation.</a>`);
  const content2 = __(`Foreman can use External service for user information and authentication. <br/>
    <a href=${getManualURL('5.7ExternalAuthentication')}>Learn more about External authentication in the documentation.</a>`);
```
Although you don't need to include all of the strings here in the dangerous html, it's just the one sentence that includes the `<em>` tags here, since the links are displayed on their own lines and not as part of the same sentence.

File: webpack/assets/javascripts/react_app/components/ConfigReports/Welcome.js [hunk 28]
```diff
@@ -0,0 +1,27 @@
...
+import { translate as __ } from '../../common/I18n';
+import EmptyState from '../common/EmptyState';
+import { getManualURL, getWikiURL } from '../../common/helpers';
+
+export const WelcomeConfigReports = () => {
+  const content = __(`If you wish to configure Puppet to forward it reports to Foreman, 
+  please follow <a href=${getManualURL(
+    '3.5.4PuppetReports'
+  )}> setting up reporting   </a> and
+  <a href=${getWikiURL('Mail_Notifications')}> e-mail reporting </a>`);
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/ConfigReports/Welcome.js]: Again something odd in the formatting here, could be eslint's fault. No need for spaces inside the `<a>` tags as well, or the underlines might display also for the spaces:
```suggestion
  const content = __(`If you wish to configure Puppet to forward it reports to Foreman, 
    please follow <a href=${getManualURL('3.5.4PuppetReports')}>setting up reporting</a> and
    <a href=${getWikiURL('Mail_Notifications')}>e-mail reporting</a>`);
```

File: app/views/report_templates/welcome.html.erb (hunk 15 above)
Comment (Ron-Lavi) [reply app/views/report_templates/welcome.html.erb:3]: I also thought it's about the time the react component takes to load into the DOM, though while testing it locally and printing the `page.body` content before the assertion, the <a> tag was there,
one thing I noticed in the dif between this PR and before, is that the <a> tag doesn't have an href,
and there's a chance capybara is looking for that, as most links have an `href`.
but let's see if that works :)

File: webpack/assets/javascripts/react_app/components/Architectures/Welcome.js (hunk 20 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/Architectures/Welcome.js]: Interesting, I like it, though maybe `sprintf` will work better in this case, though I didn't try it :D
e.g: 
```js
sprintf(__('Each entries represent a particular hardware architecture, most commonly %s)'), 'x86_64'.bold)
```

File: app/views/report_templates/welcome.html.erb (hunk 15 above)
Comment (tbrisker) [reply app/views/report_templates/welcome.html.erb:3]: good find! looks like patternfly `Button` component indeed creates an `a` tag with no `href` but rather with an event handler to handle the click.

Comment (ekohl) [reply app/views/report_templates/welcome.html.erb:3]: That sounds like a bad design in Patternfly. Is there a (good) reason they do that?

Comment (tbrisker) [reply app/views/report_templates/welcome.html.erb:3]: I assume it's because you don't always want the browser to handle the navigation, for example if you use the react router to do front-side routing.

Comment (tbrisker) [reply app/views/report_templates/welcome.html.erb:3]: Or actually it can take an `href`, it's just our `DefaultEmptyState` that implements it with a click handler - https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/common/EmptyState/DefaultEmptyState.js#L28

File: app/views/bookmarks/welcome.html.erb [hunk 29]
```diff
@@ -1,12 +1,10 @@
...
-  </div>
-  <h1><%= _('Bookmarks') %></h1>
-  <p><%= _("You don't seem to have any bookmarks.") %></p>
-  <p><%= _("Bookmarks are saved search queries. To save them, click on 'Bookmark this search' on the dropdown next to every search button.") %></p>
-  <div class="blank-slate-pf-main-action">
-      <%= link_to _('Documentation'), documentation_url("4.1.5Searching"), :rel => "external", :class => "btn btn-primary btn-lg" %>
-  </div>
-</div>
+
+<% description = _("You don't seem to have any bookmarks. Bookmarks are saved search queries. To save them, click on \'Bookmark this search\' on the dropdown next to every search button.")%>
```
Comment (tbrisker) [inline app/views/bookmarks/welcome.html.erb]: are you sure the `\'` is needed here?

File: test/integration/middleware_test.rb [hunk 30]
```diff
@@ -31,7 +31,6 @@ class MiddlewareIntegrationTest < ActionDispatch::IntegrationTest
     test 'it is added Content-Security-Policy on welcome pages' do
       Domain.stubs(:any?).returns(false)
       visit '/domains'
-      assert page.has_content? 'Learn more about this in the documentation.'
```
Comment (tbrisker) [inline test/integration/middleware_test.rb:34]: note to any other reviewer - this assertion shouldn't have been part of this test in the first place.

File: test/integration/middleware_test.rb [hunk 31]
```diff
@@ -31,7 +31,6 @@ class MiddlewareIntegrationTest < ActionDispatch::IntegrationTest
     test 'it is added Content-Security-Policy on welcome pages' do
       Domain.stubs(:any?).returns(false)
       visit '/domains'
```
Comment (tbrisker) [inline test/integration/middleware_test.rb]: This should probably also be changed to
```suggestion
      visit '/domains/help'
```

File: webpack/assets/javascripts/react_app/components/Architectures/Welcome.js [hunk 32]
```diff
@@ -0,0 +1,43 @@
...
+    url: foremanUrl('/architectures/new'),
+  };
+  const content = __(`Each entries represent a particular hardware architecture, most commonly <b>x86_64</b> or <b>i386</b>.
+  Foreman also supports the Solaris operating system family, which includes <b>sparc</b> based systems.`);
+  const description = (
+    <>
+      {__(
+        'Before you proceed to using Foreman you should provide information about one or more architectures.'
+      )}
+      {<div dangerouslySetInnerHTML={{ __html: content }} />}
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/Architectures/Welcome.js]: Are you sure you need to wrap this div in `{}`?

File: webpack/assets/javascripts/react_app/components/Architectures/Welcome.js [hunk 33]
```diff
@@ -0,0 +1,43 @@
...
+import { translate as __ } from '../../common/I18n';
+import EmptyState from '../common/EmptyState';
+import { foremanUrl } from '../../common/helpers';
+
+export const WelcomeArchitecture = ({ canCreate }) => {
+  const action = canCreate && {
+    title: __('Create Architecture'),
+    url: foremanUrl('/architectures/new'),
+  };
+  const content = __(`Each entries represent a particular hardware architecture, most commonly <b>x86_64</b> or <b>i386</b>.
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/Architectures/Welcome.js]: While we're here let's correct the English:
```suggestion
  const content = __(`Each entry represents a particular hardware architecture, most commonly <b>x86_64</b> or <b>i386</b>.
```

File: webpack/assets/javascripts/react_app/components/AuthSource/Welcome.js [hunk 34]
```diff
@@ -0,0 +1,54 @@
...
+    <>
+      {__(
+        'Foreman can use LDAP based service for user information and authentication.'
+      )}
+      {<div dangerouslySetInnerHTML={{ __html: content }} />}
+      {
+        <a href={getManualURL('4.1.1LDAPAuthentication')}>
+          {__('Learn more about LDAP authentication in the documentation.')}
+        </a>
+      }
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/AuthSource/Welcome.js]: Also here, no need to wrap components in `{}`. @laviro is it possible to enable some eslint rule to identify such cases of unneeded wrapping in jsx?

File: webpack/assets/javascripts/react_app/components/ConfigReports/Welcome.js [hunk 35]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { translate as __ } from '../../common/I18n';
+import EmptyState from '../common/EmptyState';
+import { getManualURL, getWikiURL } from '../../common/helpers';
+
+export const WelcomeConfigReports = () => {
+  const content = __(`If you wish to configure Puppet to forward it reports to Foreman,
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/ConfigReports/Welcome.js]: Another grammar correction while we're here
```suggestion
  const content = __(`If you wish to configure Puppet to forward its reports to Foreman, 
```

**Review by tbrisker — COMMENTED** [review]
Looking good! there's a couple of really small comments, feel free to do the grammar corrections in a separate commit.

File: app/views/bookmarks/welcome.html.erb (hunk 29 above)
Comment (adiabramovitch) [reply app/views/bookmarks/welcome.html.erb]: I think so, it is supposed to refer to the name of the button if I am not mistaken.

**Comment by adiabramovitch** [comment]
[test katello]

File: webpack/assets/javascripts/react_app/components/AuthSource/Welcome.js (hunk 34 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/AuthSource/Welcome.js]: sure, I think we can manage that,
opened an issue for it: https://projects.theforeman.org/issues/32416

File: app/views/bookmarks/welcome.html.erb (hunk 29 above)
Comment (Ron-Lavi) [reply app/views/bookmarks/welcome.html.erb]: maybe you can highlight it with a `<strong>Bookmark this search</strong>` tag?

**Comment by adiabramovitch** [comment]
[test katello]

**Review by Ron-Lavi — COMMENTED** [review]
overall looks great, can you do a small rebase?

**Comment by adiabramovitch** [comment]
> overall looks great, can you do a small rebase?
@laviro It is possible but I will need to write a new component. I am not sure what is the better option

**Review by tbrisker — APPROVED** [review]
Thanks @adiabramovitch ! looks great!

**Comment by ezr-ondrej** [comment]
This will need a followup so we don't loose this change in puppet relevant pages with puppet extraction: https://github.com/theforeman/foreman_puppet/issues/136
